project-chip/connectedhomeip
# Add basic support for reporting application installation status

Adding basic support for reporting application's installation status

#### Problem
In case app is installed, there's no mechanism to report the current installation status.

#### Change overview
- Added Method to App Installation Service to Retrieve App's installation Status
- Hooked the `GetInstallationStatusOfApp` with ContentAppFactory 
- Added Shell support so developers can test the method

#### Tests
1. Build the TV app using command:
`scripts/examples/gn_build_example.sh examples/tv-app/linux out/debug/`

2. Build the TV casting app using command:
`scripts/examples/gn_build_example.sh examples/tv-casting-app/linux out/debug/`

3. Start TV App
`out/debug/chip-tv-app`

4. Start TV Casting App
`out/debug/chip-tv-casting-app`

5. On TV App: Uninstall the app 
`app uninstall 65521 32769`

6. On TV Casting App: Send casting request to the tv-app using 
`cast request 0`

7. On TV App: Accepted the Cast request from the tv-casting app
`controller ux ok`

_Note that you'll see the error 13, which stands for `kAppInstallConsentPending`_

8. On TV App: Once TV app confirmed there is no Content App, try and set other app installation statuses for example:
`app setinstallerrorstatus 65521 32769 15`

9. Try to connect to the TV app again and after you enter `controller ux ok` you'll see error :
`[1716811631946] [62904:4025232] [SVR] ---- Commissioner Declaration Start ----`
`[1716811631946] [62904:4025232] [SVR] 	error code: 15`
`[1716811631946] [62904:4025232] [SVR] ---- Commissioner Declaration End ----`
